swayfmt/src/comments.rs

```diff
@@ -32,23 +83,50 @@ pub fn maybe_write_comments_from_map(
             writeln!(formatted_code)?;
         }
 
-        for comment in comments_iter {
-            // Write comments on a newline (for now). New behavior might be required
-            // to support trailing comments.
-            writeln!(
-                formatted_code,
-                "{}{}",
-                formatter.shape.indent.to_string(&formatter.config)?,
-                comment.span().as_str(),
-            )?;
+        while let Some(comment) = comments_iter.next() {
+            let newlines = collect_newlines_after_comment_span(
+                &formatter.comments_context.unformatted_code()[comment.span().end()..],
```

Comment:
This is the same as the above comment, but in the opposite direction - we're doing a `take_while()`, so I don't think this would happen. I will add some tests though just to keep this behaviour consistent + show that its working as intended